tests/functional/aws-node-sdk/test/object/websiteGet.js

```diff
@@ -0,0 +1,552 @@
+const hostname = `${bucket}.s3-website-us-east-1.amazonaws.com`;
+
+const endpoint = process.env.AWS_ON_AIR ? `${transport}://${hostname}` :
+    `${transport}://${hostname}:8000`;
+
+function putBucketWebsiteAndPutObjectRedirect(redirect, condition, key, done) {
+    const webConfig = new WebsiteConfigTester('index.html');
+    webConfig.addRoutingRule(redirect, condition);
+    s3.putBucketWebsite({ Bucket: bucket,
+        WebsiteConfiguration: webConfig }, err => {
+        if (err) {
+            done(err);
+        }
+        return s3.putObject({ Bucket: bucket,
+        Key: key,
+        ACL: 'public-read',
+        Body: fs.readFileSync(path.join(__dirname,
+            '/websiteFiles/redirect.html')),
+        ContentType: 'text/html' }, done);
+    });
+}
+
+// Note: Timeouts are set on tests with redirects to a URL as they are flaky
+// without them. If they still fail, consider increasing the timeout or using
+// mocha's this.retries method to auto-retry the test after failure.
+describe('User visits bucket website endpoint', () => {
+    const browser = new Browser();
+
+    it('should return 404 when no such bucket', done => {
+        browser.visit(endpoint, () => {
+            WebsiteConfigTester.checkHTML(browser, '404-no-such-bucket', null,
+              bucket);
+            done();
+        });
+    });
+
+    describe('with existing bucket', () => {
+        beforeEach(done => s3.createBucket({ Bucket: bucket }, done));
+
+        afterEach(done => s3.deleteBucket({ Bucket: bucket }, done));
+
+        it('should return 404 when no website configuration', done => {
+            browser.visit(endpoint, () => {
+                WebsiteConfigTester.checkHTML(browser,
+                  '404-no-such-website-configuration', null, bucket);
+                done();
+            });
+        });
+
+        describe('with existing configuration', () => {
+            beforeEach(done => {
+                const webConfig = new WebsiteConfigTester('index.html');
+                s3.putBucketWebsite({ Bucket: bucket,
+                    WebsiteConfiguration: webConfig }, err => {
+                    assert.strictEqual(err,
+                        null, `Found unexpected err ${err}`);
+                    s3.putObject({ Bucket: bucket, Key: 'index.html',
+                        ACL: 'public-read',
+                        Body: fs.readFileSync(path.join(__dirname,
+                            '/websiteFiles/index.html')),
+                        ContentType: 'text/html' },
+                        err => {
+                            assert.strictEqual(err, null);
+                            done();
+                        });
+                });
+            });
+
+            afterEach(done => {
+                s3.deleteObject({ Bucket: bucket, Key: 'index.html' },
+                err => done(err));
+            });
+
+            it('should return 405 when user requests method other than get ' +
```

Comment:
Lauren and I looked at this, it was returning 404 because the bucket and website configuration had not been created yet. (AWS will prioritize these errors over the 405 MethodNotAllowed Error). It returns 405 once we made sure these were created before the test.

We discussed trying to deliver 404 NoSuchBucket / NoSuchWebsiteConfiguration instead of 405 if the bucket / website configuration have not been created yet like AWS, but it makes more sense to us to inform the user the method is not allowed from the beginning.